### ISSUE TITLE
cmake and project tree

### ISSUE THREAD

**Issue by hrabbel** [LeMonADE-project/LeMonADE#11]
Two questions:

1) What is the use of having ${LeMonADeDirEMONADE_DIR}/tools and ${LeMonADeDirEMONADE_DIR}/extern in the top level CMakeLists.txt? Those directories don't exist...

2) Libraries are compiled into ${LeMonADeDirEMONADE_DIR}/lib , while executables are in ${LeMonADeDirEMONADE_DIR}/build/bin . Would it not be more consistent to have the executables in ${LeMonADeDirEMONADE_DIR}/bin ? Then, the build directory contains only the compiler-generated stuff

**Comment by Bondoki** [comment]
The answers :)
1) You're right: ${LEMONADE_DIR}/tools and ${LEMONADE_DIR}/extern can be removed. I think I forgot to delete this peace of code. They are intended to contain additional libs (e.g. FLTK) or other stuff. But right now, we have no additional dependencies.

2) This is more serious: I can't really figure it out, what we want to achive and what would be the best for usage of LeMonADE. First of all LeMonADE should be only a library - once compiled theres no need to modify it. On the other side, we provide basic examples how to use it - all the programs that using libLeMonADE change quite often. Actually, this is a compromise: we generate the libLeMonADE in the same root folder as we find the includes for programming. Also, we can easily link against libLeMonADE. Maybe a better approach is to put everything (include, lib, and bin) in the build directory. What do you prefer? Maybe you/we try this approach with the LeMonADE-Viewer?

**Comment by hrabbel** [comment]
Playing around with the related pull request #13 , I find I like the way it is done there. The reason is that having lib inside the build directory allows for having different compiled versions of the library in different build directories. (See also comments there)

**Comment by Bondoki** [comment]
You have already seen my proposal :) Basically, there are 2 ways of installing the libLeMonADE. There is the "classical" way by doing 

``` sh
     mkdir build
     cd build
     cmake .. [-DINSTALLDIR_LEMONADE=/place/to/put/the/lib/and/bins]
     make
```

and a now a convience script to avoid explictly user-creating build directory

``` sh
     ./configure [-DINSTALLDIR_LEMONADE=/place/to/put/the/lib/and/bins]
     make
```

Both things do the same and at least we need a build directory. The only thing I struggle is to copy all LeMonADE-headers using cmake to folder "include" in "/place/to/put/the/lib/and/bins".
Is this the way we want to do this?

**Comment by hrabbel** [comment]
Solved by pull request #20
